formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/CborElementSerializers.kt

```diff
@@ -0,0 +1,304 @@
+        when (value) {
+            is CborPrimitive -> encoder.encodeSerializableValue(CborPrimitiveSerializer, value)
+            is CborMap -> encoder.encodeSerializableValue(CborMapSerializer, value)
+            is CborList -> encoder.encodeSerializableValue(CborListSerializer, value)
+        }
+    }
+
+    override fun deserialize(decoder: Decoder): CborElement {
+        val input = decoder.asCborDecoder()
+        return input.decodeCborElement()
+    }
+}
+
+/**
+ * Serializer object providing [SerializationStrategy] and [DeserializationStrategy] for [CborPrimitive].
+ * It can only be used by with [Cbor] format an its input ([CborDecoder] and [CborEncoder]).
+ */
+internal object CborPrimitiveSerializer : KSerializer<CborPrimitive> {
+    override val descriptor: SerialDescriptor =
+        buildSerialDescriptor("kotlinx.serialization.cbor.CborPrimitive", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: CborPrimitive) {
+        val cborEncoder = encoder.asCborEncoder()
+
+        cborEncoder.encodeTags(value)
+
+        when (value) {
+            is CborNull -> encoder.encodeSerializableValue(CborNullSerializer, value)
+            is CborString -> encoder.encodeSerializableValue(CborStringSerializer, value)
+            is CborBoolean -> encoder.encodeSerializableValue(CborBooleanSerializer, value)
+            is CborByteString -> encoder.encodeSerializableValue(CborByteStringSerializer, value)
+            is CborDouble -> encoder.encodeSerializableValue(CborDoubleSerializer, value)
+            is CborNegativeInt -> encoder.encodeSerializableValue(CborIntSerializer, value)
+            is CborPositiveInt -> encoder.encodeSerializableValue(CborUIntSerializer, value)
+        }
+    }
+
+    override fun deserialize(decoder: Decoder): CborPrimitive {
+        val result = decoder.asCborDecoder().decodeCborElement()
+        if (result !is CborPrimitive) throw CborDecodingException("Unexpected CBOR element, expected CborPrimitive, had ${result::class}")
+        return result
+    }
+}
+
+/**
+ * Serializer object providing [SerializationStrategy] and [DeserializationStrategy] for [CborNull].
+ * It can only be used by with [Cbor] format an its input ([CborDecoder] and [CborEncoder]).
+ */
+internal object CborNullSerializer : KSerializer<CborNull> {
+
+    override val descriptor: SerialDescriptor =
+        buildSerialDescriptor("kotlinx.serialization.cbor.CborNull", SerialKind.ENUM)
+
+    override fun serialize(encoder: Encoder, value: CborNull) {
+        encoder.asCborEncoder().encodeTags(value)
+        encoder.encodeNull()
+    }
+
+    override fun deserialize(decoder: Decoder): CborNull {
+        decoder.asCborDecoder()
+        if (decoder.decodeNotNullMark()) {
+            throw CborDecodingException("Expected 'null' literal")
+        }
+        decoder.decodeNull()
+        return CborNull()
+    }
+}
+
+public object CborIntSerializer : KSerializer<CborNegativeInt> {
```

Comment:
Should be internal like the other specialist serializers. They are accessible on the type itself.